### PR TITLE
use single JSON config for models

### PR DESCRIPTION
**Pull request by iansimon**

### CODE REVIEW

File: music/src/music_rnn/model.ts [hunk 1]
```diff
@@ -51,14 +70,42 @@ export class MusicRNN {
...
+   * `config.json` file containing model specification from the URL directory.
+   * Does not initialize the model weights.
+   *
+   * @param url Path to a checkpoint directory.
+   * @param spec (Optional) `MusicRNNSpec` object. If undefined, will be loaded
+   * from a `config.json` file in the checkpoint directory.
+   */
+  static async fromURL(url: string, spec?: MusicRNNSpec) {
+    function modelFromSpec(spec: MusicRNNSpec) {
+      const dataConverter = data.converterFromSpec(spec.dataConverter);
```
[the remark this answers is not in the document]
Comment (iansimon) [reply music/src/music_rnn/model.ts]: Done.

File: music/README.md [hunk 2]
```diff
@@ -134,7 +134,22 @@ There are additonal flags available to reduce the size of the output by removing
 
 #### Specifying the Model Configuration
 
-TODO(iansimon): This will be dependent on how we end up setting up the model instantiation and config.
+The model configuration should be placed in a JSON file named `config.json` in the same directory as your checkpoint. This configuration file contains all the information needed (besides the weights) to instantiate and run your model: the model type and data converter specification plus optional chord encoding, auxiliary inputs, and attention length. An example `config.json` file might look like:
+
+```{
+  "type": "MusicRNN",
```
[the remark this answers is not in the document]
Comment (iansimon) [reply music/README.md]: Done.

File: music/README.md [hunk 3]
```diff
@@ -134,7 +134,23 @@ There are additonal flags available to reduce the size of the output by removing
 
 #### Specifying the Model Configuration
 
-TODO(iansimon): This will be dependent on how we end up setting up the model instantiation and config.
+The model configuration should be placed in a JSON file named `config.json` in the same directory as your checkpoint. This configuration file contains all the information needed (besides the weights) to instantiate and run your model: the model type and data converter specification plus optional chord encoding, auxiliary inputs, and attention length. An example `config.json` file might look like:
+
+```
```
Comment (adarob) [inline music/README.md]: Ad "json" after the ```

File: music/src/music_rnn/model.ts [hunk 4]
```diff
@@ -51,14 +70,49 @@ export class MusicRNN {
...
+          url, dataConverter, spec.attentionLength, chordEncoder, auxInputs);
+    }
+    if (spec) {
+      return modelFromSpec(spec);
+    } else {
+      return await fetch(`${url}/config.json`)
+          .then((response) => response.json())
+          .then((spec) => {
+            if (spec.type !== 'MusicRNN') {
+              throw new Error(
```
Comment (adarob) [inline music/src/music_rnn/model.ts]: Can you put this in modelFromSpec?

Comment (adarob) [reply music/src/music_rnn/model.ts]: Or, are you wanting to allow type to not be specified if passed via the constructor? I could go either way.

Comment (iansimon) [reply music/src/music_rnn/model.ts]: If it's passed via constructor, doesn't the compile-time check suffice?

**Pull request merged**